app/helpers/repository_helper.rb

```diff
@@ -105,10 +105,12 @@ def remove_files(files, user, repo, path: '/', txn: nil, keep_folder: true)
     end
 
     current_path = Pathname.new path
-
     current_revision = repo.get_latest_revision.revision_identifier
 
     files.each do |file_path|
+      relative_path = Pathname.new(file_path)
+      abs_path = File.join(current_path, relative_path)
+      return [false, [:file_not_exist]] unless repo.get_latest_revision.path_exists?(abs_path)
```

Comment:
The `false` is now correct when `txn` is `nil`, but not when `txn` is not nil. Look at how the return value is being computed at the bottom of this method, and try to use the same approach for this return value as well.

Reply:
Removed this check because we already checking this in repository file(git, memory and svn).
I thought it will be useful in the future If more files are accessing this method. In add_folder method we wanted to differentiate if the user trying to add folder which is already exists by adding this check and displaying 304 response code. Now we decided to display this as an error. So now 201 for success and 500 for failure with the custom failure message.